_tests/builds_test.go

```diff
@@ -1,23 +1,53 @@
 package _tests_test
 
 import (
+	"fmt"
+
 	. "github.com/onsi/ginkgo"
-	// . "github.com/onsi/gomega"
+	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("Builds", func() {
-	Context("with a deployed app", func() {
+	appName := getRandAppName()
+	Context("with a logged-in user", func() {
+		BeforeEach(func() {
+			login(url, testUser, testPassword)
+		})
 
-		It("can list app builds", func() {
-			// "deis builds:list --app=%s", app
+		Context("with no app", func() {
+			It("can create an app", func() {
+				output, err := execute("deis apps:create %s --no-remote", appName)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(output).To(ContainSubstring(fmt.Sprintf("Creating Application... done, created %s", appName)))
+			})
+			It("can deploy the app", func() {
+				output, err := execute("deis builds:create %s -a %s", "deis/example-go", appName)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(output).To(ContainSubstring("Creating build... done"))
+			})
+			It("can list app builds", func() {
+				// "deis builds:list --app=%s", app
+				output, err := execute("deis builds:list --app=%s", appName)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(output).To(SatisfyAll(
+					MatchRegexp(`[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}`)))
+			})
 		})
 
-		It("can create a build from an existing image (\"deis pull\")", func() {
-			// "deis builds:create %s --app=%s", image, app
-			// curl app
-			// `deis pull %s -a %s --procfile="worker: while true; do echo hi; sleep 3; done"`, image, app
-			// "deis ps:scale worker=1"
-			// "deis logs --app=%s", app
+		Context("with a deployed app", func() {
```

Comment:
should we remove this context if `deis builds` is already covered by "with no app"?